src/pages-sections/addAllotment/addAllotment-form.jsx

```diff
@@ -0,0 +1,165 @@
+import React from 'react';
+import { Formik } from 'formik';
+import * as yup from 'yup';
+import { Button, Card, Grid, MenuItem, TextField } from '@mui/material';
+import PropTypes from 'prop-types';
+
+const VALIDATION_SCHEMA = yup.object().shape({
+  name: yup.string().min(2).required('Name is required'),
+  bedno: yup.string().required('Bed no is required'),
+  startdate: yup.date().required('Start date is required'),
+  duedate: yup
+    .date()
+    .required('Due date is required')
+    .min(yup.ref('startdate'), 'Due date must be after start date'),
+  deposit: yup.number().required('Deposit is required'),
+  paymentmethod: yup.string().required('Payment method is required'),
+});
+const AddAllotmentForm = props => {
```

Comment:
Better to de-structure this { initialValues, handleFormSubmit } here instead of using props for only 2 fields